middleware/router.go

```diff
@@ -53,9 +53,15 @@ func NewRouter(router *gin.Engine) *gin.Engine {
 
 	// HTML Template generated pages
 	router.GET("/status", httpTemplates.StatusPageResponse(), StatusPageMiddleware())
+	router.GET("/console", ConsolePageMiddleware())
+
+	// Authentication
+	router.POST("/signup", api.Signup)
+	router.POST("/signin", api.Signin)
+	router.GET("/signout", api.Signout)
```

Comment:
Using GET for sign-out might not be the best choice. Typically, signing out changes the user's session state, which is more appropriately handled by POST or DELETE. Using GET can lead to unintended side effects, such as logging users out when they accidentally click a link or when the page is preloaded by a web crawler.